cli/command/network/connect.go

```diff
@@ -56,7 +63,23 @@ func runConnect(dockerCli *command.DockerCli, options connectOptions) error {
 		},
 		Links:   options.links.GetAll(),
 		Aliases: options.aliases,
+		DriverOpts: driverOpts,
 	}
 
 	return client.NetworkConnect(context.Background(), options.network, options.container, epConfig)
 }
+
+func convertDriverOpt(opts []string) (map[string]string, error) {
+	driverOpt := make(map[string]string)
+	for _, opt := range opts {
+		parts := strings.SplitN(opt, "=", 2)
+		if len(parts) != 2 {
+			err := fmt.Errorf("invalid key value pair format in driver options")
+			return nil, err
+		}
+		key := strings.TrimSpace(strings.ToLower(parts[0]))
+		value := strings.TrimSpace(strings.ToLower(parts[1]))
```

Comment:
For port we probably have more control over the available options (although I'd be in favour of moving validation to the daemon as well); for plugins/drivers we have no knowledge whatsoever what's supported.

It should be easy to add this back in future if we want, more difficult to remove (_"Docker 17.07 supported `opt=foObAr` and now produces an error!"_)